test/e2e/app_management_test.go

```diff
@@ -1843,6 +1844,301 @@ func TestNamespaceAutoCreation(t *testing.T) {
+				ctx.GetName(), types.JSONPatchType, []byte(`[{ "op": "replace", "path": "/spec/syncPolicy/managedNamespaceMetadata/annotations", "value": {"new":"custom-annotation"} }]`), metav1.PatchOptions{}))
+		}).
+		Sync().
+		Then().
+		Expect(Success("")).
+		Expect(Namespace(updatedNamespace, func(app *Application, ns *v1.Namespace) {
+			trackingId := ns.Annotations["argocd.argoproj.io/tracking-id"]
+
+			delete(ns.Labels, "kubernetes.io/metadata.name")
+			delete(ns.Labels, "argocd.argoproj.io/tracking-id")
+			delete(ns.Annotations, "argocd.argoproj.io/tracking-id")
+			delete(ns.Annotations, "kubectl.kubernetes.io/last-applied-configuration")
+
+			assert.Equal(t, fmt.Sprintf("%s:/Namespace:/%s", app.Name, updatedNamespace), trackingId)
+
+			assert.Equal(t, map[string]string{"new": "label"}, ns.Labels)
+			assert.Equal(t, map[string]string{"new": "custom-annotation", "argocd.argoproj.io/sync-options": "ServerSideApply=true"}, ns.Annotations)
+			assert.Equal(t, map[string]string{"new": "label"}, app.Spec.SyncPolicy.ManagedNamespaceMetadata.Labels)
+			assert.Equal(t, map[string]string{"new": "custom-annotation"}, app.Spec.SyncPolicy.ManagedNamespaceMetadata.Annotations)
+		}))
+}
+
+// Given application is set with --sync-option CreateNamespace=true
+//
+//		application --dest-namespace does not exist
+//
+//	    Verify application namespace manifest takes precedence over managedNamespaceMetadata
+func TestNamespaceAutoCreationWithMetadataAndNsManifest(t *testing.T) {
+	SkipOnEnv(t, "OPENSHIFT")
+	namespace := "guestbook-ui-with-namespace-manifest"
+	defer func() {
+		if !t.Skipped() {
+			_, err := Run("", "kubectl", "delete", "namespace", namespace)
+			assert.NoError(t, err)
+		}
+	}()
+
+	ctx := Given(t)
+	ctx.
+		SetTrackingMethod("annotation").
+		Timeout(30).
+		Path("guestbook-with-namespace-manifest").
+		When().
+		CreateFromFile(func(app *Application) {
+			app.Spec.SyncPolicy = &SyncPolicy{
+				SyncOptions: SyncOptions{"CreateNamespace=true"},
+				ManagedNamespaceMetadata: &ManagedNamespaceMetadata{
+					Labels:      map[string]string{"foo": "bar", "abc": "123"},
+					Annotations: map[string]string{"bar": "bat"},
+				}}
+		}).
+		Then().
+		Expect(NoNamespace(namespace)).
+		When().
+		AppSet("--dest-namespace", namespace).
+		Sync().
+		Then().
+		Expect(Success("")).
+		Expect(Namespace(namespace, func(app *Application, ns *v1.Namespace) {
+			assert.Empty(t, app.Status.Conditions)
+
+			trackingId := ns.Annotations["argocd.argoproj.io/tracking-id"]
+
+			delete(ns.Labels, "kubernetes.io/metadata.name")
+			delete(ns.Labels, "argocd.argoproj.io/tracking-id")
+			delete(ns.Labels, "kubectl.kubernetes.io/last-applied-configuration")
+			delete(ns.Annotations, "argocd.argoproj.io/tracking-id")
+			delete(ns.Annotations, "kubectl.kubernetes.io/last-applied-configuration")
+
+			// TODO: The tracking id is different for namespace manifests vs namespaces generated with CreateNamespace=true
```

Comment:
Just adding a comment here for adding an extra set of eyes. Currently a namespace that is managed by an Application will have a slightly different tracking id format than managed namespaces

An app that has a namespace manifest will have the format `$appName/Namespace:$nsName/$nsName` whereas a namespace that is only managed through `managedNamespaceMetadata` will have the format `$appName/Namespace:/$nsName` (i.e. the namespace only appears once). Do we to keep that difference, or should managed namespaces do the same as namespaces managed with manifests?